File: Miscellaneous Questions/Algorithms in Python/selectionSort.py

def selectionSort(list):
    for i in range(len(list)):
        # Find the minimum element in the list
        min_index = i
        for j in range(i+1, len(list)):
            if list[min_index] > list[j]:
                min_index = j
        # Swap the minimum element with the first element
        list[i], list[min_index] = list[min_index], list[i]
    return list


list = [3, 6, 8, 2, 1, 10, 4]
print("List before sorting: ", list)
sorted_list = selectionSort(list)
print("List after sorting: ", sorted_list)

File: Miscellaneous-Questions/Algorithms in JavaScript/selectionSort.js
// SelectionSort algorithm
function selectionSort(arr) {
    // loop through the array
    for (let i = 0; i < arr.length; i++) {
      // set the minimum index to the current index
      let minIndex = i;
      // loop through the rest of the array
      for (let j = i + 1; j < arr.length; j++) {
        // if the current element is smaller than the minimum element
        if (arr[j] < arr[minIndex]) {
          // set the new minimum index
          minIndex = j;
        }
      }
      // if the minimum index is not the current index
      if (minIndex !== i) {
        // swap the minimum element with the current element
        let temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
      }
    }
    // return the sorted array
    return arr;
  }
  
  // example
  let arr = [3, 5, 1, 4, 2];
  console.log(selectionSort(arr)); // [1, 2